pkg/services/contexthandler/contexthandler.go

```diff
@@ -110,6 +111,35 @@ func FromContext(c context.Context) *contextmodel.ReqContext {
 	return nil
 }
 
+// CopyWithReqContext returns a copy of the parent context with a cloned version of its RequestContext value
+func CopyReqContext(ctx context.Context) context.Context {
+	origReqCtx := FromContext(ctx)
+	if origReqCtx == nil {
+		return ctx
+	}
+
+	webCtx := &web.Context{
+		Req:  origReqCtx.Req.Clone(ctx),
+		Resp: web.NewResponseWriter(origReqCtx.Req.Method, response.CreateNormalResponse(http.Header{}, []byte{}, 0)),
+	}
+	reqCtx := &contextmodel.ReqContext{
```

Comment:
Nit. Could maybe add a Clone method to ReqContext to keep ReqContext cloning closer the definition. Or rather Copy if it's not a Clone happening

Reply:
I wasn't able to do this because it introduced circular dependencies that would be messy to remove in this PR. I made the documentation a little better, so that should help.